### PR TITLE
feat(aws/static-site): support `routes` arg

### LINKED ISSUE #4711: Static frontend + API/Function
Hi,
I'm trying to achieve a deployment of which it's similar to `StaticSite` where we only build and deploy the site's frontend on `sst deploy`, not `sst dev`, with the ability to link to an API/Function to host in the same Cdn and custom domain, like:
```
routes: {
  "/*": myStaticFrontend,
  "/api/*": myFunction.url
}
```

Is it possible / how clean is it to achieve in the current state of Ion?

### PR DESCRIPTION
**Pull request by NamesMT**
Refactored to use `Router` under-the-hood :D, tested locally with working dev, HMR, deploy.
Resolves sst/sst#4711

Tested used without routes arg. (like current)
Tested with routes from APIGatewayV2, Function

### CODE REVIEW

**Comment by NamesMT** [comment]
Example snippet of the routes usage:
```
const backend = new sst.aws.Function('Backend', {
  url: true,
  handler: 'apps/backend/src/index.handler',
})

const staticSiteDistribution = new sst.aws.StaticSite('StaticFullstack', {
  build: {
    command: 'pnpm run build --filter="frontend"',
    output: 'apps/frontend/.output/public',
  },
  routes: {
    '/api/*': backend.url,
  },
})
```

This deploys my [starter-fullstack](https://github.com/NamesMT/starter-fullstack) template to Cloudfront, with the static frontend and Hono backend served on the same domain, with working Live development access from Cloudfront's url.

File: pkg/platform/src/components/aws/static-site.ts [hunk 1]
```diff
@@ -529,64 +572,82 @@ export class StaticSite extends Component implements Link.Linkable {
...
+                },
+              };
+              newCdnArgs.origins = all([newCdnArgs.origins, s3Origin]).apply(([origins, s3Origin]) => {
+                // Replace the hard-coded '/*' origin at index 0 to our s3Origin
+                origins[0] = s3Origin
+                return origins
+              });
+
+              // @ts-expect-error I wish TS is smarter
+              Object.entries(newCdnArgs).forEach(([key, value]) => routerCdnArgs[key] = value);
```
Comment (NamesMT) [inline pkg/platform/src/components/aws/static-site.ts:641]: I forgot to apply upper-level transform here, will do tmr.

**Review by NamesMT — COMMENTED** [review]
Done.

**Comment by NamesMT** [comment]
Note: this update if merged as-is will remove and re-create the distribution for existing deploys.

**Comment by bodhihawken** [comment]
Love it when someone already has a PR for something i wanted to add. +1

**Comment by fwang** [comment]
@NamesMT refactoring to use `Router` will cause the distribution to be removed and recreated.

That said, we have plans to attach a site to a router, and you can add additional routes.

But for now, we are moving `sst/ion -> sst/sst` and unfortunately this is going to be a messy process because of github's limitations

i'm going to close this PR for now and it can be re-opened in the new `sst/sst` repo that will be ready in the next few days if still relevant

**Pull request closed without merging**